src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -4,10 +4,69 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import threading
 
 from cryptography.hazmat.bindings._openssl import ffi, lib
 
+_osrandom_engine_id = ffi.new("const char[]", b"osrandom")
+_osrandom_engine_name = ffi.new("const char[]", b"osrandom_engine")
+
+@ffi.callback("int (*)(unsigned char *, int)", error=0)
+def osrandom_rand_bytes(buf, size):
+    signed = ffi.cast("char*", buf)
+    result = os.urandom(size)
+    signed[0:size] = result
+    return 1
+
+
+@ffi.callback("int (*)(unsigned char *, int)", error=0)
+def osrandom_pseudo_rand_bytes(buf, size):
+    result = osrandom_rand_bytes(buf, size)
+    if result == 0:
+        return -1
+    else:
+        return result
+
+
+@ffi.callback("int (*)(void)")
+def osrandom_rand_status():
+    return 1
+
+
+method = ffi.new(
+    "RAND_METHOD*", dict(bytes=osrandom_rand_bytes,
+                         pseudorand=osrandom_pseudo_rand_bytes,
+                         status=osrandom_rand_status)
```

Comment:
Why not? The reasons I made it a global were:

1. @reaperhulk tells me the class is going away, so I didn't want to put more stuff on it
2. it was a global in the C code, initialized much the same way
3. `_register_osrandom_engine` already returns an integer telling you whether you registered it already or not, mirroring what the C code did previously, so having it return this would require more changes
4. it's holding references entirely to global things, so it makes sense that it live in the same scope
5. the failure mode here if registration encounters some unexpected error is that it is unused; the failure mode if I require registration to complete on the class before the reference is retained is random pointer corruption; this seemed better

So it still seems much better to me that way.  What are the arguments in *favor* of putting it on the class?